### PR TITLE
feat(edit-content): drag and drop functionality 

### LINKED ISSUE #29872: File field: Drag and drop functionality for file upload
## Parent Issue
#29691 

## Task
Develop the drag and drop functionality for file uploads in the enhanced **File field**. This should allow users to drag files from their local system and drop them into the field for immediate upload.

## Proposed Objective
User Experience

## Proposed Priority
Same as Parent Issue

## Acceptance Criteria
1. Users can drag files from their local system and drop them into the File field
2. Visual feedback is provided when a file is being dragged over the drop area
3. File upload process begins automatically when a file is dropped
4. Error handling for invalid file types or upload failures
5. Progress indicator for file uploads
6. Support for multiple file uploads if dropped simultaneously

## Assumptions & Initiation Needs
- Backend API for file uploads is available and documented
- File type restrictions and size limits are defined

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

#29872 

### Proposed Changes
* Create new component for Image & File component
* Create new state to handle drag and drop functionality
* Work in Preview component
* Upload file as dotAsset

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Screenshots

https://github.com/user-attachments/assets/90f5deb2-a416-4e7f-bfcd-db68d9d51a76

This PR fixes: #29872

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-upload-file/dot-upload-file.service.ts [hunk 1]
```diff
@@ -89,4 +89,22 @@ export class DotUploadFileService {
...
+            JSON.stringify({
+                contentlet: {
+                    file: file.name,
+                    contentType: 'dotAsset'
+                }
+            })
+        );
+
+        return this.http
+            .put<DotCMSContentlet>(`/api/v1/workflow/actions/default/fire/NEW`, formData)
```
Comment (oidacra) [inline core-web/libs/data-access/src/lib/dot-upload-file/dot-upload-file.service.ts:107]: use a constant for this endpoint

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.html [hunk 2]
```diff
@@ -1 +1,133 @@
-<p>Preview</p>
+@let previewFile = $previewFile();
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.html:1]: 🚀

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.html [hunk 3]
```diff
@@ -1 +1,133 @@
...
+        }
+        <p-button
+            (click)="removeFile.emit()"
+            styleClass="p-button-rounded "
+            data-testId="remove-button-responsive"
+            icon="pi pi-trash" />
+    </div>
+</div>
+
+<p-dialog
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.html:91]: Does this dialog exist if not used?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts [hunk 4]
```diff
@@ -1,12 +1,149 @@
...
 })
-export class DotFileFieldPreviewComponent {}
+export class DotFileFieldPreviewComponent implements OnInit {
+    readonly #dotResourceLinksService = inject(DotResourceLinksService);
+
+    $previewFile = input.required<PreviewFile>({ alias: 'previewFile' });
+    removeFile = output();
+    $showDialog = signal(false);
+
+    $metadata = computed(() => {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts:56]: Just doc.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts [hunk 5]
```diff
@@ -1,12 +1,149 @@
...
+        if (previewFile.source === 'temp') {
+            return previewFile.file.metadata;
+        }
+
+        return getFileMetadata(previewFile.file);
+    });
+
+    $content = computed(() => {
+        const previewFile = this.$previewFile();
+        if (previewFile.source === 'contentlet') {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts:67]: Does a source have a type for confirming use, `content`?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts [hunk 6]
```diff
@@ -36,37 +47,151 @@ const initialState: FileFieldState = {
...
 
             patchState(store, {
                 inputType,
-                uiMessage,
+                fieldVariable,
                 ...actions
             });
-        }
+        },
+        setUIMessage: (uiMessage: UIMessage) => {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts:91]: Just remember doc.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/utils/messages.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { MESSAGES_TYPES, UIMessagesMap } from '../models';
+
+export const UiMessageMap: UIMessagesMap = {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/utils/messages.ts:3]: I like it.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts [hunk 8]
```diff
@@ -36,37 +47,151 @@ const initialState: FileFieldState = {
...
+    maxFileSize: null,
+    fieldVariable: '',
+    previewFile: null
 };
 
 export const FileFieldStore = signalStore(
     withState(initialState),
     withComputed(({ fileStatus }) => ({
-        isEmpty: computed(() => {
+        isInit: computed(() => {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts:60]: 💪🏼

**Review by oidacra — APPROVED** [review]

File: core-web/libs/data-access/src/lib/dot-upload-file/dot-upload-file.service.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/libs/data-access/src/lib/dot-upload-file/dot-upload-file.service.ts:107]: should we use the `DotWorkflowActionsFireService` here, `newContentlet`.

**Review by hmoreras — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**